<launch>
  <!-- <node pkg="boitata_navigation" type="ground_plane_cheat" name="ground_plane_cheat" output="screen"/> -->

  <!-- fire up a nodelet manager to handle laser scan tools -->
  <node pkg="nodelet" type="nodelet" name="laser_scan_manager" args="manager" output="screen"/>


  <!-- launch laser_scan_matcher, primary output = Pose2D estimate -->
  <node pkg="nodelet" type="nodelet" name="laser_scan_matcher" args="load laser_scan_matcher/LaserScanMatcherNodelet laser_scan_manager " output="screen">
    <!-- inputs -->
    <remap from="scan" to="laser_scan"/>
    <remap from="odom" to="odom"/>
    <remap from="laser" to="rp_lidar_A1"/>

    <!-- outputs -->
    <param name="publish_tf" value="false"/> 
    <param name="publish_pose" value="true"/>
    <remap from="pose2D" to="LaserScan/pose2D"/>
    <!-- <remap from="odom_out" to="LaserScan/odom"/> -->

    <!-- motion prediction for faster convergence -->
    <param name="use_imu" value="true"/>
    <param name="use_odom" value="true"/>
    <param name="fixed_frame" value="odom"/>
    <param name="base_frame" value="base_link"/>

    <!-- KeyFrame params (TL;DR= only update when there's motion) -->
    <param name="kf_dist_linear" value="0.05"/>
    <param name="kf_dist_angular" value="0.1"/>
  </node>

  <!-- this node just converts the above Pose2D estimate to an odom topic for use in EKF -->
  <node pkg="dhex_localization" type="pose2D_to_odometry.py" name="Pose2D_to_Odometry" output="screen"/>

</launch>